CRM/Dashlet/Page/GettingStarted.php

```diff
@@ -0,0 +1,152 @@
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ *
+ * @package CRM
+ * @copyright CiviCRM LLC (c) 2004-2015
+ * $Id$
+ *
+ */
+
+/**
+ * Main page for getting started dashlet
+ */
+class CRM_Dashlet_Page_GettingStarted extends CRM_Core_Page {
+
+  const CHECK_TIMEOUT = 5;
+  const CACHE_DAYS = 5;
+  const GETTING_STARTED_URL = 'https://alert.civicrm.org/welcome?prot=1&ver={ver}&uf={uf}&sid={sid}&lang={lang}&co={co}';
+
+  /**
+   * Define tokens available for getting started
+   */
+  static $_tokens = array(
+    'crmurl' => array(
+      'configbackend' => 'civicrm/admin/configtask',
+    ),
+  );
+
+  /**
+   * Get the final, usable URL string (after interpolating any variables)
+   *
+   * @return FALSE|string
+   */
+  public function gettingStartedUrl() {
+    // Note: We use "*default*" as the default (rather than self::GETTING_STARTED_URL) so that future
+    // developers can change GETTING_STARTED_URL without needing to update {civicrm_setting}.
+    $url = CRM_Core_BAO_Setting::getItem(CRM_Core_BAO_Setting::SYSTEM_PREFERENCES_NAME, 'gettingStartedUrl', NULL, '*default*');
+    if ($url === '*default*') {
+      $url = self::GETTING_STARTED_URL;
+    }
+    return CRM_Utils_System::evalUrl($url);
+  }
+
+  /**
+   * List gettingStarted page as dashlet.
+   */
+  public function run() {
+    $context = CRM_Utils_Request::retrieve('context', 'String', $this, FALSE, 'dashlet');
+    $this->assign('context', $context);
+
+    $this->assign('gettingStarted', $this->_gettingStarted());
+
+    return parent::run();
+  }
+
+  /**
+   * Load gettingStarted page from cache.
+   * Refresh cache if expired
+   *
+   * @return array
+   */
+  private function _gettingStarted() {
+    // Fetch data from cache
+    $cache = CRM_Core_DAO::executeQuery("SELECT data, created_date FROM civicrm_cache
+      WHERE group_name = 'dashboard' AND path = 'gettingStarted'");
+    if ($cache->fetch()) {
+      $expire = time() - (60 * 60 * 24 * self::CACHE_DAYS);
+      // Refresh data after CACHE_DAYS
+      if (strtotime($cache->created_date) < $expire) {
+        $new_data = $this->_getHtml($this->gettingStartedUrl());
+        // If fetching the new html was successful, return it
+        // Otherwise use the old cached data - it's better than nothing
+        if ($new_data) {
+          return $new_data;
+        }
+      }
+      return unserialize($cache->data);
+    }
+    return $this->_getHtml($this->gettingStartedUrl());
```

Comment:
For future reference... it's usually nice to have the cache-get and cache-set in the same function. There's nothing obvious in the name `getHtml()` which would indicate that `getHtml()` also updates the cache.

Typical idiom for code touching a cache:

```php
$value = getFromCache()
if (!$value) {
  $value = computeValue();
  saveToCache($value);
}
return $value;
```